### DISCUSSION TITLE
Restoring files from usb to home fails

### DISCUSSION THREAD

**Discussion by ghost** [rustic-rs/rustic#637] [Things are not working as expected - is this a bug?]
I run this: `rustic -r /run/media/username/usb/data/bkp restore latest:/home/username .`

This gives me constantly the error: `creating special file failed.`

What I want is to restore the backup from my usb (/run/media/username/usb/data/bkp) to my /home/username in the computer (preferably not overwriting the files that are already in my /home/username).

And is there any documentation or videos for rustic?

**Comment by aawsome** [comment]
> This gives me constantly the error: `creating special file failed.`

This should actually be just a warning. You seem to have some special file (e.g. device file or fifo) which cannot be created by the restore command. Maybe this is due to permissions. Did you try to restore using the root permissions?

> What I want is to restore the backup from my usb (/run/media/username/usb/data/bkp) to my /home/username in the computer (preferably not overwriting the files that are already in my /home/username).

`restore` by default doesn't remove existing files if they are not present in the snapshot (there is `--delete`, though), but does overwrite existing files if they differ from the snapshot content.

> And is there any documentation or videos for rustic?

This is open, see #302

**Comment by ghost** [comment]
Now I do not have any error but restore does not work. I have tried deleting one file of my /home/user that I have backed up previously in my usb. Then I have tried these commands but they have not restored the file (that was previously backed up). 
```
rustic -r "/run/media/user/usb/data/bkp" restore latest:/home/user .
rustic -r "/run/media/user/usb/data/bkp" restore latest:/home/user/exactpathofdeletedfile .
```
The output I get: 

```
[00:00:00] getting latest snapshot...     ████████████████████████████████████████     11/11              
[00:00:00] reading index...               ████████████████████████████████████████         11/11              
[00:00:00] collecting file information... ⠉                                                                  
[WARN] Note: additional entries exist in destination
[00:00:00] collecting file information...                                                                     
Files:  0 to restore, 0 unchanged, 0 verified, 0 to modify, 35225 additional
Dirs:   33 to restore, 0 to modify, 1369 additional
[INFO] total restore size: 0 B
[INFO] all file contents are fine.
[00:00:01] setting metadata...                                                                                
[INFO] restore done.
```

**Reply by aawsome** [reply]
Seems there is no file in the snapshot subpath you are referenceing. You can list the files using e.g. `rustic ls --recursive latest:/home/user`

**Reply by ghost** [reply]
I have tried your command and the path exists. I think I will wait to the documentation. Thanks.
